docs/examples/general/data_loading/webdataset.ipynb

```diff
@@ -0,0 +1,514 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #16.                buffer.append(next(generator))
</pre><p>Nitpick: Not sure how expensive is to query the length of the list over and over again, but it can maybe optimized by doing something like:</p><pre class="ql-syntax" spellcheck="false">to_fill = initial_fill - len(buffer) if len(buffer) &lt; initial_fill else 0
for _ in range(to_fill):
    buffer.append(next(generator))

</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/3153/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Reply:
<p>just checked, not more expensive than this</p><p>(source: https://stackoverflow.com/a/1115401)</p>